.github/actions/generate-commit-list-v1/src/run.ts

```diff
@@ -1,34 +1,64 @@
 import type { Core, Github } from './types'
 import getRawCommitList from './getRawCommitList'
 import getParsedCommitList from './getParsedCommitList'
-import doesBranchExist from './doesBranchExist'
+import doesBranchOrTagExist from './doesBranchOrTagExist'
 
 export default async function run(core: Core, github: Github) {
   try {
-    const base = core.getInput('base', { required: true })
-    const head = core.getInput('head', { required: true })
+    const base = core.getInput('base')
+    const head = core.getInput('head')
+    const tag = core.getInput('tag')
 
-    const [doesBaseExist, doesHeadExist] = await Promise.all([
-      doesBranchExist(base),
-      doesBranchExist(head)
-    ])
-
-    core.info(`Checking if ${base} exists...`)
-    if (!doesBaseExist) {
-      core.setFailed(`The base branch ${base} does not exist.`)
+    if (!tag && (!base || !head)) {
```

Comment:
Can apply [De Morgan's Law](https://brilliant.org/wiki/de-morgans-laws/) to simplify the statement a bit

```suggestion
    if (!(base && head || tag)) {
```